src/packages/@app/pages/TeamExplorerPage/TeamExplorerPage.ViewModel.ts

```diff
@@ -505,4 +506,43 @@ export class TeamExplorerPageViewModel {
       );
     }
   };
+
+  /**
+   * Delete the workspace and update the local DB as per changes
+   * Updates workspace, team, and tab repository.
+   * Also updates active workspace state, if active workspace is deleted
+   * @param workspace - workspace document
+   * @returns - A promise that resolves when the delete workspace is complete.
+   */
+  public handleDeleteWorkspace = async (
+    workspace: WorkspaceDocument,
+  ): Promise<MakeRequestResponse> => {
+    const isActiveWorkspace =
+      await this.workspaceRepository.checkActiveWorkspace(workspace._id);
+    const workspaces = await this.workspaceRepository.getWorkspacesDocs();
+    if (isActiveWorkspace && workspaces.length === 1) {
+      notifications.error(
+        "Failed to delete the last workspace. Please create a new workspace before deleting this workspace.",
+      );
+    }
+    const response = await this.workspaceService.deleteWorkspace(workspace._id);
+    if (response.isSuccessful) {
+      await this.workspaceRepository.deleteWorkspace(workspace._id);
+      if (isActiveWorkspace) {
+        await this.workspaceRepository.activateInitialWorkspace();
+      }
+      await this.teamRepository.removeWorkspaceFromTeam(
```

Comment:
It does not much matter which workspace are we activating because we are activating workspace from our own, previously we are giving user the option to select which workspace he wants to activate, but as we are doing it in our own, we should follow a straightforward approach, to set initial workspace as active.